src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryProgrammingExerciseParticipationResource.java

```diff
@@ -225,6 +226,24 @@ public ResponseEntity<Map<String, String>> getFilesWithContent(@PathVariable Lon
         });
     }
 
+    /**
+     * GET /repository/{participationId}/file-names: Gets the file names of the repository
+     *
+     * @param participationId participation of the student/template/solution
+     * @return the ResponseEntity with status 200 (OK) and a set of file names
+     */
+    @GetMapping(value = "/repository/{participationId}/file-names", produces = MediaType.APPLICATION_JSON_VALUE)
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<Set<String>> getFileNames(@PathVariable Long participationId) {
```

Comment:
Same as above: It looks like you did not add any Unit/Integration tests for this endpoint. Please consider adding one.

Reply:
Implemented